src/ecs/resources/input.rs

```diff
@@ -195,18 +208,18 @@ impl InputHandler {
     }
 
     /// Returns a vector containing pressed down mouse buttons.
-    pub fn pressed_mouse_buttons(&self) -> &SmallVec<[MouseButton; 16]> {
-        &self.pressed_mouse_buttons
+    pub fn pressed_mouse_buttons(&self) -> &[MouseButton] {
+        &*self.pressed_mouse_buttons
     }
 
     /// Returns a vector containing mouse buttons pressed on this frame.
-    pub fn down_mouse_buttons(&self) -> &SmallVec<[MouseButton; 8]> {
-        &self.down_mouse_buttons
+    pub fn down_mouse_buttons(&self) -> &[MouseButton]  {
+        &*self.down_mouse_buttons
     }
 
     /// Returns a vector containing mouse buttons released on this frame.
-    pub fn released_mouse_buttons(&self) -> &SmallVec<[MouseButton; 8]> {
-        &self.released_mouse_buttons
+    pub fn released_mouse_buttons(&self) -> &[MouseButton] {
```

Comment:
I would find it more consistent to return an iterator everywhere or a slice everywhere. Since the latter seems to be impossible, let's go for the former. You can just use `slice::Iter` here.